name 'Response' is not defined

On Flask 2, I get this error every time I run a query.  I have solved the issue only changing the code of Flask-Injector but I am not confident with the impact of the change I have done. 

This error is present when you run the unit test suite.
The fix I propose broke the test `test_forward_references_work` :(. 

Version:

```
flask: 2.0.1
flask-injector: 0.13.0
flask-restx: 0.5.1
injector: 0.18.4
```

```
127.0.0.1 - - [09/Sep/2021 10:23:38] "GET /swaggerui/favicon-16x16.png HTTP/1.1" 500 -
Error on request:
Traceback (most recent call last):
  File "/home/project/.venv/lib/python3.8/site-packages/injector/__init__.py", line 1176, in _infer_injected_bindings
    bindings = get_type_hints(callable, include_extras=True)
  File "/home/project/.venv/lib/python3.8/site-packages/typing_extensions.py", line 1924, in get_type_hints
    hint = typing.get_type_hints(obj, globalns=globalns, localns=localns)
  File "/usr/lib/python3.8/typing.py", line 1264, in get_type_hints
    value = _eval_type(value, globalns, localns)
  File "/usr/lib/python3.8/typing.py", line 270, in _eval_type
    return t._evaluate(globalns, localns)
  File "/usr/lib/python3.8/typing.py", line 518, in _evaluate
    eval(self.__forward_code__, globalns, localns),
  File "<string>", line 1, in <module>
    import importlib.util
NameError: name 'Response' is not defined
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/project/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 319, in run_wsgi
    execute(self.server.app)
  File "/home/project/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 308, in execute
    application_iter = app(environ, start_response)
  File "/home/project/.venv/lib/python3.8/site-packages/flask/app.py", line 2088, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/project/.venv/lib/python3.8/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.handle_exception(e)
  File "/home/project/.venv/lib/python3.8/site-packages/flask_restx/api.py", line 672, in error_router
    return original_handler(e)
  File "/home/project/.venv/lib/python3.8/site-packages/flask/app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/project/.venv/lib/python3.8/site-packages/flask/app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/project/.venv/lib/python3.8/site-packages/flask_restx/api.py", line 672, in error_router
    return original_handler(e)
  File "/home/project/.venv/lib/python3.8/site-packages/flask/app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/project/.venv/lib/python3.8/site-packages/flask/app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/project/.venv/lib/python3.8/site-packages/flask_injector/__init__.py", line 89, in wrapper
    return injector.call_with_injection(callable=fun, args=args, kwargs=kwargs)
  File "/home/project/.venv/lib/python3.8/site-packages/injector/__init__.py", line 1010, in call_with_injection
    bindings = get_bindings(callable)
  File "/home/project/.venv/lib/python3.8/site-packages/injector/__init__.py", line 1163, in get_bindings
    callable, _infer_injected_bindings(callable, only_explicit_bindings=look_for_explicit_bindings)
  File "/home/project/.venv/lib/python3.8/site-packages/injector/__init__.py", line 1178, in _infer_injected_bindings
    raise _BindingNotYetAvailable(e)
```

I track the root cause in `wrap_fun` that is supposed to managed this sort of error. On Flask, 

```python
  if hasattr(fun, '__call__') and not isinstance(fun, type):
      try:
          type_hints = get_type_hints(fun)
      except (AttributeError, TypeError):
          # Some callables aren't introspectable with get_type_hints,
          # let's assume they don't have anything to inject. The exception
          # types handled here are what I encountered so far.
          # It used to be AttributeError, then https://github.com/python/typing/pull/314
          # changed it to TypeError.
          wrap_it = False
      except NameError:
          wrap_it = True
```

On Flask, the error is due to the usage of typehint with in `flask/scaffold.py`

```
if t.TYPE_CHECKING:
    from .wrappers import Response

[...]
    def send_static_file(self, filename: str) -> "Response":
```

The function get_type_hints raise a `NameError` exception in this context.

# Proposition of fix

I suppose the existance of NameError has a reason but moving it in Non wrapping condition solve the issue I have. I didn't find the reason of the existence of this condition.

```python
  if hasattr(fun, '__call__') and not isinstance(fun, type):
      try:
          type_hints = get_type_hints(fun)
      except (AttributeError, TypeError, NameError):
          # Some callables aren't introspectable with get_type_hints,
          # let's assume they don't have anything to inject. The exception
          # types handled here are what I encountered so far.
          # It used to be AttributeError, then https://github.com/python/typing/pull/314
          # changed it to TypeError.
          wrap_it = False
```

## More information

 the issue with get_type_hints and Forward Ref

* https://github.com/python/typing/issues/508
* https://bugs.python.org/issue43463

The above PR #71 is the proposed fix from @FabienArcellier, and as he said, now only the case `test_forward_references_work` failed.
But I have no idea how to fix here, more wisdom is indeed very needed. 🥲 

```
FAIL: flask_injector.tests.test_forward_references_work
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/czhang/workspace/vendor/flask_injector/ve3.9/lib/python3.9/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/Users/czhang/workspace/vendor/flask_injector/flask_injector/tests.py", line 412, in test_forward_references_work
    eq_(response.data.decode(), 'Hello World')
AssertionError: '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">\n<title>500 Internal Server Error</title>\n<h1>Internal Server Error</h1>\n<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>\n' != 'Hello World'
-------------------- >> begin captured logging << --------------------
flask_injector.tests: ERROR: Exception on / [GET]
Traceback (most recent call last):
  File "/Users/czhang/workspace/vendor/flask_injector/ve3.9/lib/python3.9/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/czhang/workspace/vendor/flask_injector/ve3.9/lib/python3.9/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/czhang/workspace/vendor/flask_injector/ve3.9/lib/python3.9/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/czhang/workspace/vendor/flask_injector/ve3.9/lib/python3.9/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
TypeError: index() missing 1 required positional argument: 'x'
--------------------- >> end captured logging << ---------------------
```

--- a/flask_injector/__init__.py
+++ b/flask_injector/__init__.py
@@ -62,15 +62,13 @@ def wrap_fun(fun: T, injector: Injector) -> T:
     if hasattr(fun, '__call__') and not isinstance(fun, type):
         try:
             type_hints = get_type_hints(fun)
-        except (AttributeError, TypeError):
+        except (AttributeError, TypeError, NameError):
             # Some callables aren't introspectable with get_type_hints,
             # let's assume they don't have anything to inject. The exception
             # types handled here are what I encountered so far.
             # It used to be AttributeError, then https://github.com/python/typing/pull/314
             # changed it to TypeError.
             wrap_it = False
-        except NameError:
-            wrap_it = True
         else:
             type_hints.pop('return', None)
             wrap_it = type_hints != {}

